File: Day_8/cycle_detection.java
 static boolean hasCycle(SinglyLinkedListNode head) {
        SinglyLinkedListNode temp=head;
        boolean flag=false;
        while(temp.next!=null)
        {
            
            if(temp.data==-1)
            {
                flag=true;
                break;
            }
            else{
                temp.data=-1;
            }
            temp=temp.next;
        }
        return flag;
    }

File: Day_8/cycle_detection.js
function hasCycle(head) {
    var slowPtr = head;
    var fastPtr = head.next;
    if(head==null || head.next == null){
        return false;
    }
    while(fastPtr){
        if(slowPtr == twoStepNode){
            return true;
        }
        slowPtr = slowPtr.next;
        fastPtr = fastPtr.next.next;
    }
    return false;
}